src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,104 @@
+package storage;
+
+import exceptions.DukeException;
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Task;
+import tasks.Todo;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+public class Storage {
+    private String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads the tasks stored in the file and returns them as a list of tasks.
+     * @return list of tasks stored in the text document
+     * @throws DukeException err i not v sure
+     */
+    public ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            String content = new String(Files.readAllBytes(Paths.get(filePath)), "UTF-8");
```

Comment:
You can use Scanner object to read file ie. Scanner sc = new Scanner(new File(filePath)); , which gives you access to nextLine() and hasNext() methods.